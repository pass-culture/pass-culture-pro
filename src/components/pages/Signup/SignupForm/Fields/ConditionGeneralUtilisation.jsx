import React from "react"
import { Field } from 'react-final-form'

function ConditionGeneralUtilisation() {
  return (
    <Field
      name="cgu_ok"
    >
      {({ input }) => (
        <div
          className="field field-checkbox cgu-field"
        >
          <label
            className="required"
            htmlFor="user-cgu_ok"
          >
            <input
              {...input}
              aria-describedby="user-cgu_ok-error"
              className="input"
              id="user-cgu_ok"

              required=""
              type="checkbox"
              value=""
            />
            {'J’ai lu et j’accepte les '}
            <a
              href="https://docs.passculture.app/textes-normatifs"
              id="accept-cgu-link"
              rel="noopener noreferrer"
              target="_blank"
            >
              {'Conditions Générales d’Utilisation'}
            </a>
          </label>
        </div>
      )}
    </Field>
  )
}

export default ConditionGeneralUtilisation
