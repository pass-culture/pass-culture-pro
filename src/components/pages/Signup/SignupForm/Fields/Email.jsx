import { Field } from "react-final-form"
import React from "react"
import isRequired from '../validators/isRequired'
import isValidEmail from '../validators/isValidEmail'
import { composeValidators } from 'react-final-form-utils'

function Email() {
  return (
    <Field
      name="email"
      validate={composeValidators(isRequired, isValidEmail)}
    >
      {({ input, meta }) => {
        return (
          <div className="field field-email with-subtitle">
            <label
              className="label"
              htmlFor="user-email"
            >
              <h3 className="required">
                {'Adresse e-mail'}
              </h3>
              <p>
                {'... pour se connecter et récupérer son mot de passe en cas d’oubli :'}
              </p>
            </label>
            <div className="control">
              <input
                aria-describedby="user-email-error"
                autoComplete="email"
                className="input is-normal"
                id="user-email"
                name="email"
                placeholder="nom@exemple.fr"
                required
                type="email"
                {...input}
              />
            </div>
            {meta.error && meta.touched && (
              <ul
                className="help is-danger"
                id="user-email-error"
              >
                <p
                  className="help is-danger columns"
                  id="user-email-error"
                >
                  <span className="column is-narrow is-vcentered">
                    <img
                      alt="Attention"
                      src="/icons/picto-warning.svg"
                    />
                  </span>
                  <span className="column is-paddingless is-narrow">
                    {meta.error}
                  </span>
                </p>
              </ul>
            )}
          </div>
        )
      }}
    </Field>)
}

export default Email
