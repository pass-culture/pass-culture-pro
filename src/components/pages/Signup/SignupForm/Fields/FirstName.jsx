import { Field } from "react-final-form"
import React from "react"
import isRequired from '../validators/isRequired'
import { composeValidators } from 'react-final-form-utils'

function FirstName() {
  return (
    <Field
      name="firstName"
      validate={composeValidators(isRequired)}
    >
      {({ input, meta }) => {
        return (
          <div className="field field-firstName">
            <label
              className="label"
              htmlFor="user-firstName"
            >
              <h3 className="required">
                {'Prénom'}
              </h3>
            </label>
            <div className="control">
              <input
                aria-describedby="user-email-error"
                className="input is-normal"
                id="user-firstName"
                name="firstName"
                placeholder="Mon prénom"
                required
                {...input}
              />
            </div>
            {meta.error && meta.touched && (
              <ul
                className="help is-danger"
                id="user-email-error"
              >
                <p
                  className="help is-danger columns"
                  id="user-email-error"
                >
                  <span className="column is-narrow is-vcentered">
                    <img
                      alt="Attention"
                      src="/icons/picto-warning.svg"
                    />
                  </span>
                  <span className="column is-paddingless is-narrow">
                    {meta.error}
                  </span>
                </p>
              </ul>
            )}
          </div>
        )
      }}
    </Field>)
}

export default FirstName
