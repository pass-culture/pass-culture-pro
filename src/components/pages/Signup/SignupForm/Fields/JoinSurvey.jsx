import React from "react"
import { Field } from 'react-final-form'

function JoinSurvey() {
  return (
    <Field
      name="contact"
    >
      {({ input }) => (
        <div className="field field-checkbox">
          <label
            className=""
            htmlFor="user-contact_ok"
          >
            <input
              {...input}
              aria-describedby="user-contact_ok-error"
              className="input"
              id="user-contact_ok"
              type="checkbox"
              value=""
            />
            {'J’accepte d\'être contacté par e-mail pour donner mon avis sur le pass Culture'}
          </label>
        </div>
      )}
    </Field>
  )
}

export default JoinSurvey
