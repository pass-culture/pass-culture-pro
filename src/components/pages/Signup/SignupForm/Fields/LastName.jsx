import { Field } from "react-final-form"
import React from "react"
import isRequired from '../validators/isRequired'
import { composeValidators } from 'react-final-form-utils'

function LastName() {
  return (
    <Field
      name="lastName"
      validate={composeValidators(isRequired)}
    >
      {({ input, meta }) => {
        return (
          <div className="field field-lastName">
            <label
              className="label"
              htmlFor="user-lastName"
            >
              <h3 className="required">
                {'Nom'}
              </h3>
            </label>
            <div className="control">
              <input
                aria-describedby="user-email-error"
                className="input is-normal"
                id="user-lastName"
                name="lastName"
                placeholder="Mon nom"
                required
                {...input}
              />
            </div>
            {meta.error && meta.touched && (
              <ul
                className="help is-danger"
                id="user-email-error"
              >
                <p
                  className="help is-danger columns"
                  id="user-email-error"
                >
                  <span className="column is-narrow is-vcentered">
                    <img
                      alt="Attention"
                      src="/icons/picto-warning.svg"
                    />
                  </span>
                  <span className="column is-paddingless is-narrow">
                    {meta.error}
                  </span>
                </p>
              </ul>
            )}
          </div>
        )
      }}
    </Field>)
}

export default LastName
