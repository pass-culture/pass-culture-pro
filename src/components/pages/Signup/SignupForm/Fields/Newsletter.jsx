import React from "react"

function Newsletter() {
  return (
    <div className="field field-checkbox">
      <label
        className=""
        htmlFor="user-newsletter_ok"
      >
        <input
          aria-describedby="user-newsletter_ok-error"
          className="input"
          id="user-newsletter_ok"
          name="newsletter_ok"
          type="checkbox"
          value=""
        />
        {'Je souhaite recevoir les actualit√©s du pass Culture'}
      </label>
    </div>
  )
}

export default Newsletter
