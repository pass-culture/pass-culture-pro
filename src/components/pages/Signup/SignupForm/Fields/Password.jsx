import { Field } from "react-final-form"
import React from "react"
import isRequired from '../validators/isRequired'
import { composeValidators } from 'react-final-form-utils'

function Password() {
  return (
    <Field
      name="password"
      validate={composeValidators(isRequired)}
    >
      {({ input, meta }) => {
        return (
          <div className="field field-password with-subtitle">
            <label
              className="label"
              htmlFor="user-password"
            >
              <h3 className="required">
                {'Mot de passe'}
              </h3>
              <p>
                {'... pour se connecter'}
              </p>
            </label>
            <div className="control">
              <input
                aria-describedby="user-email-error"
                className="input is-normal"
                id="user-password"
                name="password"
                placeholder="Mon mot de passe"
                required
                {...input}
              />
            </div>
            {meta.error && meta.touched && (
              <ul
                className="help is-danger"
                id="user-email-error"
              >
                <p
                  className="help is-danger columns"
                  id="user-email-error"
                >
                  <span className="column is-narrow is-vcentered">
                    <img
                      alt="Attention"
                      src="/icons/picto-warning.svg"
                    />
                  </span>
                  <span className="column is-paddingless is-narrow">
                    {meta.error}
                  </span>
                </p>
              </ul>
            )}
          </div>
        )
      }}
    </Field>)
}

export default Password
