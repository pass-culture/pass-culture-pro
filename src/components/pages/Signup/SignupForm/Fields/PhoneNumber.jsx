import { Field } from "react-final-form"
import React from "react"
import isRequired from '../validators/isRequired'

function PhoneNumber() {
  return (
    <Field
      name="phoneNumber"
      validate={isRequired}
    >
      {({ input, meta }) => {
        return (
          <div className="field field-phoneNumber with-subtitle">
            <label
              className="label"
              htmlFor="user-phoneNumber"
            >
              <h3 className="required">
                {'Téléphone'}
              </h3>
              <p>
                {'... utilisé uniquement par l\'équipe du pass Culture :'}
              </p>
            </label>
            <div className="control">
              <input
                aria-describedby="user-phoneNumber-error"
                autoComplete="email"
                className="input is-normal"
                id="user-phoneNumber"
                name="email"
                placeholder="nom@exemple.fr"
                required
                {...input}
              />
            </div>
            {meta.error && meta.touched && (
              <ul
                className="help is-danger"
                id="user-phoneNumber-error"
              >
                <p
                  className="help is-danger columns"
                  id="user-phoneNumber-error"
                >
                  <span className="column is-narrow is-vcentered">
                    <img
                      alt="Attention"
                      src="/icons/picto-warning.svg"
                    />
                  </span>
                  <span className="column is-paddingless is-narrow">
                    {meta.error}
                  </span>
                </p>
              </ul>
            )}
          </div>
        )
      }}
    </Field>)
}

export default PhoneNumber
