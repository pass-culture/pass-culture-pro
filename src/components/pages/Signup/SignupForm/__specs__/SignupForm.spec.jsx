import React from 'react'
import SignupForm from '../SignupForm'
import { shallow } from 'enzyme'
import { NavLink } from 'react-router-dom'
import Email from '../Fields/Email'
import Password from '../Fields/Password'
import LastName from '../Fields/LastName'
import FirstName from '../Fields/FirstName'
import PhoneNumber from '../Fields/PhoneNumber'
import Siren from '../Fields/Siren'
import Newsletter from '../Fields/Newsletter'
import JoinSurvey from '../Fields/JoinSurvey'
import ConditionGeneralUtilisation from '../Fields/ConditionGeneralUtilisation'

describe('src | components | pages | Signup', () => {
  let props

  beforeEach(() => {
    props = {
      errors: [],
      offererName: 'super structure',
      patch: {},
    }
  })

  describe('render', () => {
    it('should render a disabled submit button when required inputs are not filled', () => {
      // Given
      props.pristine = false
      props.invalid = true

      // when
      const wrapper = shallow(<SignupForm {...props} />)

      // then
      const submitButton = wrapper.find('button[type="submit"]')
      expect(submitButton.prop('disabled')).toBe(true)
    })

    it('should render a Field component for email with the right props', () => {
      // when
      const wrapper = shallow(<SignupForm {...props} />)

      // then
      const field = wrapper.find(Email)
      expect(field).toHaveLength(1)
    })

    it('should render a Field component for password with the right props', () => {
      // when
      const wrapper = shallow(<SignupForm {...props} />)

      // then
      const field = wrapper.find(Password)
      expect(field).toHaveLength(1)
    })

    it('should render a Field component for lastname with the right props', () => {
      // when
      const wrapper = shallow(<SignupForm {...props} />)

      // then
      const field = wrapper.find(LastName)
      expect(field).toHaveLength(1)
    })

    it('should render a Field component for firstname with the right props', () => {
      // when
      const wrapper = shallow(<SignupForm {...props} />)

      // then
      const field = wrapper.find(FirstName)
      expect(field).toHaveLength(1)
    })

    it('should render a Field component for phone number with the right props', () => {
      // when
      const wrapper = shallow(<SignupForm {...props} />)

      // then
      const field = wrapper.find(PhoneNumber)
      expect(field).toHaveLength(1)
    })

    it('should render a Field component for siren with the right props', () => {
      // when
      const wrapper = shallow(<SignupForm {...props} />)

      // then
      const field = wrapper.find(Siren)
      expect(field).toHaveLength(1)
    })

    it('should render a Field component for newsletter agreement with the right props', () => {
      // when
      const wrapper = shallow(<SignupForm {...props} />)

      // then
      const field = wrapper.find(Newsletter)
      expect(field).toHaveLength(1)
    })

    it('should render a Field component for contact agreement with the right props', () => {
      // when
      const wrapper = shallow(<SignupForm {...props} />)

      // then
      const field = wrapper.find(JoinSurvey)
      expect(field).toHaveLength(1)
    })

    it('should render a Field component for cgu agreement with the right props', () => {
      // when
      const wrapper = shallow(<SignupForm {...props} />)

      // then
      const field = wrapper.find(ConditionGeneralUtilisation)
      expect(field).toHaveLength(1)
    })

    it('should render a NavLink component with the right props', () => {
      // when
      const wrapper = shallow(<SignupForm {...props} />)

      // then
      const navLink = wrapper.find(NavLink)
      expect(navLink).toHaveLength(1)
      expect(navLink.prop('className')).toBe('button is-secondary')
      expect(navLink.prop('to')).toBe('/connexion')
    })

    it('should render a SubmitButton component with the right props', () => {
      // when
      const wrapper = shallow(<SignupForm {...props} />)

      // then
      const submitButton = wrapper.find('button[type="submit"]')
      expect(submitButton).toHaveLength(1)
      expect(submitButton.prop('className')).toBe('button is-primary is-outlined')
      expect(submitButton.prop('type')).toBe('submit')
      expect(submitButton.text()).toBe('Cr√©er')
    })
  })
})
