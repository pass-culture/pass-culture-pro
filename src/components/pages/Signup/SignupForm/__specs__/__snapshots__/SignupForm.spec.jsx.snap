// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | pages | Signup | SignupForm should match snapshot 1`] = `ShallowWrapper {}`;
